diff --git a/node_modules/@dnd-kit/core/dist/core.esm.js b/node_modules/@dnd-kit/core/dist/core.esm.js
index 3cc765e..12722f1 100644
--- a/node_modules/@dnd-kit/core/dist/core.esm.js
+++ b/node_modules/@dnd-kit/core/dist/core.esm.js
@@ -1153,7 +1153,7 @@ class KeyboardSensor {
 
   attach() {
     this.handleStart();
-    this.windowListeners.add(EventName.Resize, this.handleCancel);
+    //this.windowListeners.add(EventName.Resize, this.handleCancel);
     this.windowListeners.add(EventName.VisibilityChange, this.handleCancel);
     setTimeout(() => this.listeners.add(EventName.Keydown, this.handleKeyDown));
   }
@@ -1441,7 +1441,7 @@ class AbstractPointerSensor {
       this.listeners.add(events.cancel.name, this.handleCancel);
     }
 
-    this.windowListeners.add(EventName.Resize, this.handleCancel);
+    //this.windowListeners.add(EventName.Resize, this.handleCancel);
     this.windowListeners.add(EventName.DragStart, preventDefault);
     this.windowListeners.add(EventName.VisibilityChange, this.handleCancel);
     this.windowListeners.add(EventName.ContextMenu, preventDefault);
